core/src/main/java/com/graphhopper/routing/AbstractBidirectionEdgeCHNoSOD.java

```diff
@@ -131,8 +132,8 @@ protected int getIncomingEdge(SPTEntry entry) {
 
     @Override
     protected int getTraversalId(RoutingCHEdgeIteratorState edge, int origEdgeId, boolean reverse) {
-        int baseNode = getOtherNode(origEdgeId, edge.getAdjNode());
-        return GHUtility.createEdgeKey(baseNode, edge.getAdjNode(), origEdgeId, reverse);
+        EdgeIteratorState origEdge = graph.getBaseGraph().getEdgeIteratorState(origEdgeId, edge.getAdjNode());
+        return origEdge.getEdgeKey(reverse);
```

Comment:
Using `getOtherNode` is better for performance here, I agree these `getTraversalId` methods are a bit twisted any hopefully this can be improved.

Reply:
Then again we should not need this at all and the edge key should be known already. But then we need to store the first/last orig edge key (not id) for the edge-based CH shortcuts (but maybe this is better anyway...).